irremote/irremote.go

```diff
@@ -0,0 +1,173 @@
+package irremote // import "tinygo.org/x/drivers/irremote"
+
+import (
+	"machine"
+	"time"
+)
+
+// NEC protocol references
+// https://www.sbprojects.net/knowledge/ir/nec.php
+// https://techdocs.altium.com/display/FPGA/NEC+Infrared+Transmission+Protocol
+// https://simple-circuit.com/arduino-nec-remote-control-decoder/
+
+// Client callback function for IR commands
+// 'code' is the raw NEC format code. 'addr' & 'cmd' are the NEC decoded address and command codes respectively
+// See the NEC protocol reference links above for more information
+type IRCallback func(code uint32, addr uint16, cmd uint8, repeat bool)
+
+// Public API for an IR receiver device
+type IRReceiverDevice interface {
```

Comment:
Is there another device that implements this interface? Just wondering if it is better to wait until there is another before using interface here?

Reply:
Yes, you're probably right. My thinking was that extending to cover other protocols (e.g. Sony SIRC, Phillips RC5/RC6 etc.) may result in additional devices in the future, but that's probably premature for now, especially on a pre-1.0 release. 

On a similar note I considered renaming the `New()` factory/constructor to `NewNEC()` for the same reason, thinking maybe it should be explicit and also reserving the generic `New()` for a potential 'automatic' protocol sensing device in the future? Again, I guess that's premature, although it does raise the question of whether we should strive to maintain backwards source compatibility in iterative versions prior to a 1.0 release? My opinion would be 'no' but happy to go witht he consensus here. Any guidance?

Reply:
> whether we should strive to maintain backwards source compatibility in iterative versions prior to a 1.0 release

yes, unless we have a good reason to change. In which case better to change everything in a single release if possible.